File: src/main/java/com/demonwav/statcraft/magic/ColorCode.java
/*
 * StatCraft Bukkit Plugin
 *
 * Copyright (c) 2016 Kyle Wood (DemonWav)
 * https://www.demonwav.com
 *
 * MIT License
 */

package com.demonwav.statcraft.magic;


public enum ColorCode {
    WHITE       ((byte)0),
    ORANGE      ((byte)1),
    MAGENTA     ((byte)2),
    LIGHT_BLUE  ((byte)3),
    YELLOW      ((byte)4),
    LIME        ((byte)5),
    PINK        ((byte)6),
    GRAY        ((byte)7),
    LIGHT_GRAY  ((byte)8),
    CYAN        ((byte)9),
    PURPLE      ((byte)10),
    BLUE        ((byte)11),
    BROWN       ((byte)12),
    GREEN       ((byte)13),
    RED         ((byte)14),
    BLACK       ((byte)15);

    private byte code;

    ColorCode(byte code) {
        this.code = code;
    }

    public byte getCode() {
        return code;
    }

    public static ColorCode fromCode(byte code) {
        for (ColorCode colorCode : values()) {
            if (code == colorCode.getCode())
                return colorCode;
        }
        return null;
    }
}

File: src/main/kotlin/com/demonwav/statcraft/magic/ColorCode.kt
/*
 * StatCraft Bukkit Plugin
 *
 * Copyright (c) 2016 Kyle Wood (DemonWav)
 * https://www.demonwav.com
 *
 * MIT License
 */

package com.demonwav.statcraft.magic

enum class ColorCode(val code: Byte) {

    WHITE       (0.toByte()),
    ORANGE      (1.toByte()),
    MAGENTA     (2.toByte()),
    LIGHT_BLUE  (3.toByte()),
    YELLOW      (4.toByte()),
    LIME        (5.toByte()),
    PINK        (6.toByte()),
    GRAY        (7.toByte()),
    LIGHT_GRAY  (8.toByte()),
    CYAN        (9.toByte()),
    PURPLE      (10.toByte()),
    BLUE        (11.toByte()),
    BROWN       (12.toByte()),
    GREEN       (13.toByte()),
    RED         (14.toByte()),
    BLACK       (15.toByte());

    companion object {
        @JvmStatic
        fun fromCode(code: Byte): ColorCode? {
            for (colorCode in values()) {
                if (code == colorCode.code)
                    return colorCode
            }
            return null
        }
    }
}
